### PR TITLE
Add new rule `no-obscure-array-access`

### LINKED ISSUE #787: add new rule: no-obscure-hbs
Originally opened in https://github.com/ember-template-lint/ember-cli-template-lint/issues/437
I have some handlebars syntax in my application that "works", but is "wrong".
[Twiddle](https://ember-twiddle.com/07c76e285823a864ecfd1a59dc754705?openFiles=templates.application.hbs%2C)
```hbs
<p>some user name {{this.list.[0].name}}</p>
```

locks and rwjblue proposed adding a template rule to warn/error before any deprecations. ( see [ember-cli discord](https://discordapp.com/channels/480462759797063690/486548111221719040) on 2/06/19)

### PR DESCRIPTION
**Pull request by GoygovRustam**
Adding a rule to discourage usage of the following syntax in the template `{{list.[0]}}`

Fixes #787.

### CODE REVIEW

File: docs/rule/no-obscure-hbs.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# no-obscure-hbs
+
+Using obscrure expressions `{{list.[1].name}}` is descouraged and might be deprecated soon
+
+This rule suggests usage of the Ember's getter instead.
```
Comment (bmish) [inline docs/rule/no-obscure-hbs.md:5]: Should we really suggest Ember's `get` function? I thought Ember is moving away from using its custom `get` function:

https://github.com/ember-cli/eslint-plugin-ember/blob/master/docs/rules/no-get.md

File: docs/rule/no-obscure-hbs.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# no-obscure-hbs
```
Comment (bmish) [inline docs/rule/no-obscure-hbs.md:1]: We should name this something specific to this issue like `no-obscure-array-access`.

File: docs/rule/no-obscure-hbs.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+{{foo bar (get @list '1')}}
+```
+
+```hbs
+<Foo @bar={{get list '0'}} />
+```
+
+## References
+
+- [Github Issue #787](https://github.com/ember-template-lint/ember-template-lint/issues/787)
```
Comment (bmish) [inline docs/rule/no-obscure-hbs.md:39]: We shouldn't link to the rule ticket. Any information from the rule ticket should be included directly in this documentation file. And we can link to any relevant Ember documentation.

File: package-lock.json [hunk 4]
```diff
@@ -0,0 +1,9722 @@
+{
```
Comment (bmish) [inline package-lock.json:1]: This file should be removed. We use yarn.

**Review by bmish — CHANGES_REQUESTED** [review]

File: docs/rule/no-obscure-hbs.md (hunk 1 above)
Comment (GoygovRustam) [reply docs/rule/no-obscure-hbs.md:5]: I meant "get"  helper not the "get" function.  I'll update this description.

File: package-lock.json (hunk 4 above)
Comment (GoygovRustam) [reply package-lock.json]: removed

File: docs/rule/no-obscure-hbs.md (hunk 3 above)
Comment (GoygovRustam) [reply docs/rule/no-obscure-hbs.md]: removed ticket link and added link to the discord discussion

File: docs/rule/no-obscure-hbs.md (hunk 2 above)
Comment (GoygovRustam) [reply docs/rule/no-obscure-hbs.md]: renamed

**Pull request closed without merging**